compiler/rustc_query_system/src/query/job.rs

```diff
@@ -591,10 +591,14 @@ pub(crate) fn report_cycle<'a>(
         err.span_note(span, &format!("...which requires {}...", query.description));
     }
```

Comment:
I'm intrigued if the iteration above doesn't need to be reverted to be correct 🤔 (no need to address now)